### DISCUSSION TITLE
`$TMPPREFIX` is not an environment variable?

### DISCUSSION THREAD

**Discussion by eamonburns** [ohmyzsh/ohmyzsh#12016] [Help] [answered]
(I am not entirely sure this is the right place to ask this question, but this is the best place I have found so far)
I am trying to access the temporary directory using an environment variable (for compatibility), so, I run `echo $TMPPREFIX` this prints `/tmp/zsh` that is good. I then try it in a bash script, and it is undefined. I then (in the shell) type `export TMPPREFIX=${TMPPREFIX}` and run the bash script again and it works. Why is `TMPPREFIX` a shell variable and not an environment variable? Is there any way to access it from a bash script? Is there a better way of getting the path to the temporary directory? (Don't know why, but the `TMP` and `TMPDIR` environment variables are not defined either)

**Comment by orzomaxx** [comment, marked as answer]
`TMPPREFIX` is zsh-specific variable

**Reply by eamonburns** [reply]
yes, I know. Is there a way to access it from a script? If not, is there a better way of getting the path to the temporary directory?

**Reply by carlosala** [reply]
You can both execute the script with `zsh` (if is POSIX-compliant) or, in case it's bash/sh, you can always execute the script with `TMPPREFIX=$TMPPREFIX ./script.sh`.
